Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs

```diff
@@ -45,36 +45,40 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
             var localPath = httpContext.Request.Path.ToString();
             var baseUrlPrefix = serverConfigurationManager.GetNetworkConfiguration().BaseUrl;
 
-            if (!string.IsNullOrEmpty(baseUrlPrefix))
+            if (string.IsNullOrEmpty(localPath)
+                || string.Equals(localPath, "/", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, "/web", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, "/web/", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, baseUrlPrefix + "/", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, baseUrlPrefix + "/web", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(localPath, baseUrlPrefix + "/web/", StringComparison.OrdinalIgnoreCase)
+                || !localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
+               )
             {
-                var startsWithBaseUrl = localPath.StartsWith(baseUrlPrefix, StringComparison.OrdinalIgnoreCase);
-
-                if (!startsWithBaseUrl
-                    && (localPath.Equals("/health", StringComparison.OrdinalIgnoreCase)
-                        || localPath.Equals("/health/", StringComparison.OrdinalIgnoreCase)))
+                // Redirect health endpoint
+                if (string.Equals(localPath, "/health", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(localPath, "/health/", StringComparison.OrdinalIgnoreCase))
                 {
                     _logger.LogDebug("Redirecting /health check");
                     httpContext.Response.Redirect(baseUrlPrefix + "/health");
                     return;
                 }
 
-                if (!startsWithBaseUrl
-                    || localPath.Length == baseUrlPrefix.Length
-                    // Local path is /baseUrl/
-                    || (localPath.Length == baseUrlPrefix.Length + 1 && localPath[^1] == '/'))
+                // Always redirect back to the default path if the base prefix is invalid or missing
+                _logger.LogDebug("Normalizing an URL at {LocalPath}", localPath);
+
+                if (string.IsNullOrWhiteSpace(baseUrlPrefix) || baseUrlPrefix[^1] != '/')
```

Comment:
This is always true https://github.com/jellyfin/jellyfin/blob/90174f68e2b477e613d756d78cbfd89c2a33f30a/Jellyfin.Networking/Configuration/NetworkConfiguration.cs#L64